AWSWindowsHelpers/public/ARN/Get-AWSWindowsHelpersARNDetail.ps1

```diff
@@ -0,0 +1,128 @@
+<#
+.Synopsis
+   Returns the details of an AWS Resource from the ARN
+.DESCRIPTION
+   Returns the details of an AWS Resource from the ARN
+.EXAMPLE
+   Get-AWSWindowsHelpersARNDetail -arn "arn:aws:iam::123456789012:server-certificate/2017_wild_example_com"
+.INPUTS
+   arn - ARN of the AWS certificate
+.OUTPUTS
+   Returns a object with the following properties
+   Partition - The partition the resource is in
+   Service - The AWS Service
+   Region - The AWS region
+   AccountID - AWS account number
+   ResourceDetail - The resource ARN detail
+   ResourceType - The type of AWS resource
+.FUNCTIONALITY
+   Retrieves from a ARN the details of the AWS service
+#>
+Function Get-AWSWindowsHelpersARNDetail
+{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$true)]
+        [ValidatePattern('^arn:*.:*.:*.:*.:*.')]
+        $arn      
+    )
+    $arnSplit = $arn -split ":"
+
+    $arnDetail = @{
+        partition = $arnSplit[1]
+        service = $arnSplit[2]
+        region = $arnSplit[3]
+        accountid = $arnSplit[4]
+    }
+
+    $resourceDetail = $arnSplit[5..$arnSplit.Length] -join ":"
+    $arnDetail.Add("resourcedetail",$resourceDetail)
+    $resourceARN = $resourceDetail -split "/"
+    $resourceList= @()
+
+    if($arnSplit.Length -le 6 -and $($resourceARN.length -eq 1) -or $arnDetail.service -eq 's3')
+    {   # Deal with ARN in format "arn:partition:service:region:account-id:resource"
+        if($($resourceARN.length -gt 1))
+        { # Handle S3 resources
```

Comment:
Can you clarify this comment? I assume this populates `$arndetail.resource` with the... bucketname?

Reply:
Yes, I'll try and clarify it. This section of code deals with S3 bucket ARNs and ARNs which only have a resource element at the end. E.g. `arn:aws:codepipeline:us-east-1:123456789012:MyDemoPipeline` or `arn:aws:codestar:us-east-1:123456789012:my-first-project`